packages/jest-environment-puppeteer/src/global.js

```diff
@@ -19,7 +19,7 @@ export async function setup() {
   const config = await readConfig()
   browser = await puppeteer.launch(config.launch)
   mkdirp.sync(DIR)
-  fs.writeFileSync(WS_ENDPOINT_PATH, browser.wsEndpoint())
+  fs.writeFileSync(WS_ENDPOINT_PATH + ((process.ppid) ? process.pid : ""), browser.wsEndpoint())
```

Comment:
In global mode we are in ppid? No? I really don't understand the logic.

Reply:
We check for ppid in global mode because if it is not defined the user's Node doesn't support it and we are falling back to the old ws-endpoint naming scheme. We only ever use pid in global mode because it is the parent process if we have multiple test files or the same process for a single test file.